.drone.yml

```diff
@@ -3,31 +3,77 @@ kind: pipeline
 name: pipeline1
 
 steps:
+- name: restore-cache
+  image: maven:3.6.3-adoptopenjdk-8
+  failure: ignore
+  volumes:
+  - name: m2-cache
+    path: /.m2
+  commands:
+  - ./.ci/drone-io.sh restore-maven-cache
+
 - name: checkstyle-and-sevntu
   image: maven:3.6.3-adoptopenjdk-8
+  environment:
+    MAVEN_OPTS: "-Dmaven.repo.local=/.m2"
+  volumes:
+  - name: m2-cache
+    path: /.m2
   commands:
   - ./.ci/travis/travis.sh checkstyle-and-sevntu
 
 - name: spotbugs-and-pmd
   image: maven:3.6.3-adoptopenjdk-8
+  environment:
+    MAVEN_OPTS: "-Dmaven.repo.local=/.m2"
+  volumes:
+  - name: m2-cache
+    path: /.m2
   commands:
   - ./.ci/travis/travis.sh spotbugs-and-pmd
 
+volumes:
+- name: m2-cache
+  temp: {}
```

Comment:
why we need this ?

Reply:
This is a temporary volume to keep state between stages of pipeline. E.g. container1 starts and restores state to /.m2, then container2 (likely another docker image) will start and run actual work, it won't out of box have all changes from container1. Thats where temporary volumes are handy.
More: https://docs.drone.io/pipeline/docker/syntax/volumes/temporary/